api/integrations/github/models.py

```diff
@@ -57,21 +58,26 @@ class Meta:
                     "repository_name",
                 ],
                 name="unique_repository_data",
+                condition=models.Q(deleted_at__isnull=True),
             )
         ]
+        ordering = ("id",)
 
     @hook(BEFORE_DELETE)
     def delete_feature_external_resources(
         self,
     ) -> None:
         from features.feature_external_resources.models import (
             FeatureExternalResource,
+            ResourceType,
         )
 
         FeatureExternalResource.objects.filter(
             feature_id__in=self.project.features.values_list("id", flat=True),
             type__in=[
-                FeatureExternalResource.ResourceType.GITHUB_ISSUE,
-                FeatureExternalResource.ResourceType.GITHUB_PR,
+                ResourceType.GITHUB_ISSUE,
+                ResourceType.GITHUB_PR,
             ],
+            # Filter by url containing the repository owner and name
+            url__contains=f"{self.repository_owner}/{self.repository_name}",

```

Comment:
The match criteria isn't good enough. If a repository owner is named `foo` it would also match to `examplefoo`. you need to bound the region of the `url__contains` so that you're always matching the right object.

Reply:
I'm using regex instead of contains